### ISSUE TITLE
Render content within <template> tag

### ISSUE THREAD

**Issue by pwasem** [PolymerElements/iron-list#139]
Hi,

I am trying to build a `my-list` element which uses `iron-list` and add some extra features.
User should be able to provide **items** as well as an **item-template** from the outside as follows

```
<my-list items="{{items}}">
  <div item>
    <p>Name: <span>[[item.name]]</span></p>
    <p>Age: <span>[[item.age]]</span></p>
  </div>
</my-list>
```

Within `my-list` the `<content>`  should be selected into the `iron-list`'s `<template>` which is'n working as follows:

```
<link rel="import" href="../../bower_components/polymer/polymer.html">
<link rel="import" href="../../bower_components/iron-list/iron-list.html">
<dom-module id="my-list">
  <template>
    <style>
      :host {
        display: block;
      }
    </style>
    <div flex>
      <iron-list id="list" items="{{items}}">
        <template>
          <content id="content" select="[item]"></content>
        </template>
      </iron-list>
    </div>
  </template>
  <script>
    (function () {
      'use strict';
      Polymer({
        'is': 'my-list',
        'properties': {
          'items': {
            'type': Array,
            'value': [],
            'notify': true
          }
        }
      });
    })();
  </script>
</dom-module>

```

I this even possible? 
Maybe it can be done programmatically if `<content>` is placed outside `<template>`

``` javascript
function onAttached() {

  let _this = this;

  let content = _this.getContentChildren('#content');
  let userTemplate = content[0];

  let listTemplate = Polymer.dom(_this.$.list).querySelector('template');
  Polymer.dom(listTemplate).appendChild(userTemplate);
  Polymer.dom.flush();
  _this.$.list._render();
}
```

Thanks!

**Comment by blasten** [comment]
This is a known limitation. I will ask @kevinpschaaf to see if we would support this use case.

**Comment by pwasem** [comment]
Thank you for your response. I think this is a common use case, especially in the context web component composition.

**Comment by Zecat** [comment]
I tried another approach

```html
<my-list list-items="[[data]]">
  <template>
    <p>Name: <span>[[item.name]]</span></p>
    <p>Age: <span>[[item.age]]</span></p>
  </template>
</my-list>
```
```html
<link rel="import" href="../../bower_components/polymer/polymer.html">
<link rel="import" href="../../bower_components/iron-list/iron-list.html">
<dom-module id="my-list">
  <template>
    <style>
      :host {
        display: block;
      }
    </style>
    <div flex>
      <iron-list items="[[listItems]]">
        <content></content>
      </iron-list>
    </div>
  </template>
  <script>
    (function () {
      'use strict';
      Polymer({
        'is': 'my-list',
        'properties': {
          'listItems': {
            'type': Array,
            'value': []
          }
        }
      });
    })();
  </script>
</dom-module>
```

But requires a fix on https://github.com/PolymerElements/iron-list/blob/master/iron-list.html#L793
```html
this._userTemplate = Polymer.dom(this).querySelector('template');
```
to replace with
```html
this._userTemplate = this.querySelector('template');
```
It unlocks a great feature !

**Comment by pwasem** [comment]
Hi @Zecat, I have just tested it and it works! Thanks.
I got some minor styling issues, but data binding just works as expected.

Pull request?

**Comment by Zecat** [comment]
It works but on reflection, we should use the DOM API anyway. From [local dom doc](https://www.polymer-project.org/1.0/docs/devguide/local-dom.html) :
> Shady DOM requires you to use the Polymer DOM API when manipulating DOM from JavaScript. This interface covers most of the common DOM methods and properties, and is compatible with both shady DOM and native shadow DOM. 
…
Polymer provides a custom API for manipulating DOM such that local DOM and light DOM trees are properly maintained.
…
All DOM manipulation must use this API, as opposed to DOM API directly on nodes.

So instead :
```javascript
this._userTemplate = Polymer.dom(this).querySelector('template') || this.queryEffectiveChildren('template');
```
In addition I think it improves performance, but in return, the template isn't searched recursively in content 's childrens, so content has to be at the root of the inserted content, I mean below worked but not anymore :
```html
<my-list list-items="[[data]]">
  <div>
    <template>
      <p>Name: <span>[[item.name]]</span></p>
      <p>Age: <span>[[item.age]]</span></p>
    </template>
  </div>
</my-list>
```

**Comment by timove** [comment]
+1

**Comment by atzt** [comment]
+1 This would be a great feature

**Comment by colintoal** [comment]
+1 - or let me extend iron-list with new elements and behaviours.

**Comment by blasten** [comment]
Fixed in [v1.4.0](https://github.com/PolymerElements/iron-list/releases/tag/v1.4.0).
